src/main/cluster-manager.ts

```diff
@@ -49,6 +49,8 @@ export class ClusterManager {
         // we need to swap path prefix so that request is proxied to kube api
         req.url = req.url.replace(`/${clusterId}`, apiKubePrefix)
       }
+    } else if (req.headers["x-cluster-id"]) {
+      cluster = clusterStore.getById(req.headers["x-cluster-id"].toString())
```

Comment:
So is it that solution to overcome `fetch`/`header` problem? or it's totally different?

Reply:
Yes, this is a workaround for `fetch` host header issue.